src/main/java/com/artgallery/infrastructure/ArtRepositoryJson.java

```diff
@@ -45,12 +45,11 @@ public void addPainting(Painting painting) {
     }
 
     @Override
-    public void updatePainting(Painting painting) {
+    public void updatePainting(UUID id) {
         List<Painting> paintings = getPaintings();
-        paintings.remove(painting);
-        paintings.add(painting);
-        JSONArray jsonArray = ArtMapper.mapPaintings(paintings);
-        writeToFile(jsonArray);
+        paintings.remove(getPainting(id, paintings));
+        Painting painting = new Painting(getPainting(id, paintings).getId());
```

Comment:
1) znajdz Painting dla podanego ID i usuń go z listy, to widzę, że tak chcesz zrobić
2) dodaj do listy przekazany Painting poprzez metodę addPainting którą już masz gotową :)